### PR TITLE
Clean up AddressBook dependencies

### PR DESCRIPTION
**Pull request by woojiahao**
Free at last

### CODE REVIEW

**Comment by woojiahao** [comment]
Proposing to merge regardless of failing code coverage for following reason:

The original AddressBook had a code coverage of 75.26%: https://app.codecov.io/github/se-edu/addressbook-level3/commit/da647a31d3f265287dbaf3cac6adc8a7272be68c

The reason why this PR is failing is because the combination of UniCa$h + AddressBook files inflated the test coverage as there were way more files. However, after deleting the AddressBook dependencies, there's no way that the test coverage can remain the same. I have added as much coverage to existing files (thus a -0.38% difference now) and will continue to add more as more commands/behavior is added

File: src/test/java/seedu/address/model/UniCashTest.java [hunk 1]
```diff
@@ -111,19 +112,21 @@ public void toStringMethod() {
     public void equals() {
         // same object -> returns true
         UniCash transactionList = new UniCashBuilder().build();
-        assertTrue(transactionList.equals(transactionList));
+        assertEquals(transactionList, transactionList);
```
Comment (jamz903) [inline src/test/java/seedu/address/model/UniCashTest.java:115]: see above comments!

**Review by jamz903 — COMMENTED** [review]
Saw that you have changed usage of `assertFalse` and `assertTrue` to `assertNotEquals` and `assertEquals` respectively. Commented on one of the files about it, here are the other test files that are also affected:

```
EditTransactionCommandTest.java
FindCommandTest.java
CategoryTest.java
DateTimeTest.java
LocationTest.java
NameTest.java
TransactionListTest.java
TransactionNameContainsKeywordsPredicateTest.java
TransactionTest.java
TypeTest.java
```

File: src/main/java/seedu/address/model/Model.java [hunk 2]
```diff
@@ -5,17 +5,15 @@
...
 /**
  * The API of the Model component.
  */
 public interface Model {
-    /** {@code Predicate} that always evaluate to true */
-    Predicate<Person> PREDICATE_SHOW_ALL_PERSONS = unused -> true;
-
-    /** {@code Predicate} that always evaluate to true */
+    /**
+     * {@code Predicate} that always evaluate to true
```
Comment (sp4ce-cowboy) [inline src/main/java/seedu/address/model/Model.java:15]:
```suggestion
     * {@code Predicate} that always evaluates to true
```

File: src/main/java/seedu/address/model/Model.java [hunk 3]
```diff
@@ -38,54 +36,13 @@ public interface Model {
...
-     * Replaces the given person {@code target} with {@code editedPerson}.
-     * {@code target} must exist in the address book.
-     * The person identity of {@code editedPerson} must not be the same as another existing person in the address book.
-     */
-    void setPerson(Person target, Person editedPerson);
-
     /**
      * Replaces the given transaction {@code target} with {@code editedTransaction}.
      * {@code target} must exist in the transaction list.
      * The transaction identity of {@code editedTransaction} must not be the same as another existing transaction.
```
Comment (sp4ce-cowboy) [inline src/main/java/seedu/address/model/Model.java:42]: not really relevant to this PR review but just to clarify, we are defining transaction "identity" with Transaction.equals() right? does this mean that for e.g. a user won't be able to buy the same item in the same shop within the same hour? do we wanna (later on) consider implementing UID() instead?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -20,29 +19,25 @@
...
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
 
-    private final AddressBook addressBook;
     private final UniCash uniCash;
     private final UserPrefs userPrefs;
-    private final FilteredList<Person> filteredPersons;
     private final FilteredList<Transaction> filteredTransactions;
 
     /**
      * Initializes a ModelManager with the given addressBook, userPrefs and uniCash.
```
Comment (sp4ce-cowboy) [inline src/main/java/seedu/address/model/ModelManager.java:27]:
```suggestion
     * Initializes a ModelManager with the given userPrefs and UniCash.
```

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -71,57 +66,9 @@ public void setGuiSettings(GuiSettings guiSettings) {
 
     //=========== AddressBook ================================================================================
```
Comment (sp4ce-cowboy) [inline src/main/java/seedu/address/model/ModelManager.java:67]: I think we can remove this whole line right

Comment (sp4ce-cowboy) [inline src/main/java/seedu/address/model/ModelManager.java:67]:
```suggestion
   
```

File: src/main/java/seedu/address/model/ModelManager.java [hunk 6]
```diff
@@ -71,57 +66,9 @@ public void setGuiSettings(GuiSettings guiSettings) {
...
-    }
-
-    @Override
-    public void setPerson(Person target, Person editedPerson) {
-        requireAllNonNull(target, editedPerson);
-
-        addressBook.setPerson(target, editedPerson);
-    }
 
     //=========== Filtered Person List Accessors =============================================================
```
Comment (sp4ce-cowboy) [inline src/main/java/seedu/address/model/ModelManager.java:70]: this line too
```suggestion

```

File: src/test/java/seedu/address/AppParametersTest.java [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/AppParametersTest.java:5]: not an issue at all, but just wondering is there any reason why we're switching from assertTrue to assertNotEquals?

File: src/test/java/seedu/address/AppParametersTest.java [hunk 8]
```diff
@@ -51,25 +51,27 @@ public void equals() {
...
         // different config path -> returns false
         AppParameters otherAppParameters = new AppParameters();
         otherAppParameters.setConfigPath(Paths.get("configPath"));
-        assertFalse(appParameters.equals(otherAppParameters));
+        assertNotEquals(appParameters, otherAppParameters);
     }
 
     private static class ParametersStub extends Application.Parameters {
-        private Map<String, String> namedParameters = new HashMap<>();
+        private final Map<String, String> namedParameters = new HashMap<>();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/AppParametersTest.java:74]: good idea to make namedParameters final

File: src/test/java/seedu/address/commons/core/ConfigTest.java [hunk 9]
```diff
@@ -20,8 +24,33 @@ public void toStringMethod() {
...
+        assertNotEquals(otherConfig, defaultConfig);
+
+        otherConfig = new Config();
+        otherConfig.setUserPrefsFilePath(Path.of("invalid_path.txt"));
+        assertNotEquals(otherConfig, defaultConfig);
     }
 
+    @Test
+    public void hashCode_sameConfig_returnsSameHashCode() {
+        var first = new Config();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/commons/core/ConfigTest.java:43]: maybe we can change var type to Config? 
```suggestion
        Config first = new Config();
```

File: src/test/java/seedu/address/commons/core/ConfigTest.java [hunk 10]
```diff
@@ -20,8 +24,33 @@ public void toStringMethod() {
...
+        var second = new Config();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/commons/core/ConfigTest.java:44]:
```suggestion
        Config second = new Config();
```

File: src/test/java/seedu/address/commons/core/ConfigTest.java [hunk 11]
```diff
@@ -20,8 +24,33 @@ public void toStringMethod() {
...
+        assertEquals(first.hashCode(), second.hashCode());
+    }
+
+    @Test
+    public void hashCode_differentConfig_returnsDifferentHashCode() {
+        var first = new Config();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/commons/core/ConfigTest.java:50]:
```suggestion
        Config first = new Config();
```

File: src/test/java/seedu/address/commons/core/ConfigTest.java [hunk 12]
```diff
@@ -20,8 +24,33 @@ public void toStringMethod() {
...
+        var second = new Config();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/commons/core/ConfigTest.java:51]:
```suggestion
        Config second = new Config();
```

File: src/test/java/seedu/address/commons/util/FileUtilTest.java [hunk 13]
```diff
@@ -20,4 +28,17 @@ public void isValidPath() {
...
     }
 
+    @Test
+    public void isFileExists_nonExistentFile_returnsFalse() {
+        assertFalse(FileUtil.isFileExists(Path.of("invalid")));
+    }
+
+    @Test
+    public void createFile_existingFile_earlyReturns() throws IOException {
+        var path = temporaryFolder.resolve("tempfile");
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/commons/util/FileUtilTest.java:38]:
```suggestion
        Path path = temporaryFolder.resolve("tempfile");
```

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 14]
```diff
@@ -84,13 +87,21 @@ public void execute_storageThrowsAdException_throwsCommandException() {
...
     @Test
-    public void getFilteredTransactionList_modifyList_throwsUnsupportedOperationException() {
-        assertThrows(UnsupportedOperationException.class, () -> logic.getFilteredTransactionList().remove(0));
+    public void getGuiSettings() {
+        assertEquals(model.getGuiSettings(), new GuiSettings());
+    }
+
+    @Test
+    public void setGuiSettings() {
+        var localModel = new ModelManager();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:101]:
```suggestion
        ModelManager localModel = new ModelManager();
```

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 15]
```diff
@@ -84,13 +87,21 @@ public void execute_storageThrowsAdException_throwsCommandException() {
...
+        var guiSettings = new GuiSettings(15, 15, 15, 15);
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:102]:
```suggestion
        GuiSettings guiSettings = new GuiSettings(15, 15, 15, 15);
```

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 16]
```diff
@@ -22,8 +22,8 @@
  * Contains integration tests (interaction with the Model) for {@code FindCommand}.
  */
 public class FindCommandTest {
-    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs(), new UniCash());
-    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs(), new UniCash());
+    private final Model model = new ModelManager(new UniCash(), new UserPrefs());
+    private final Model expectedModel = new ModelManager(new UniCash(), new UserPrefs());
     @Test
     public void equals() {
         TransactionNameContainsKeywordsPredicate firstPredicate =
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:29]: Now that we've deleted the original NameContainsKeywordsPredicate, do we want to change the unsightly TransactionNameContainsKeywordsPredicate back or keep it the same for now?

File: src/test/java/seedu/address/model/UserPrefsTest.java [hunk 17]
```diff
@@ -18,4 +25,20 @@ public void setUniCashFilePath_nullPath_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> userPrefs.setUniCashFilePath(null));
     }
 
+    @Test
+    public void equals() {
+        var userPrefs = new UserPrefs();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/model/UserPrefsTest.java:30]:
```suggestion
        UserPrefs userPrefs = new UserPrefs();
```

File: src/test/java/seedu/address/model/UserPrefsTest.java [hunk 18]
```diff
@@ -18,4 +25,20 @@ public void setUniCashFilePath_nullPath_throwsNullPointerException() {
...
+        assertEquals(userPrefs, userPrefs);
+
+        assertFalse(userPrefs.equals(5));
+
+        var changedGuiSettings = new UserPrefs();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/model/UserPrefsTest.java:35]:
```suggestion
        UserPrefs changedGuiSettings = new UserPrefs();
```

File: src/test/java/seedu/address/model/UserPrefsTest.java [hunk 19]
```diff
@@ -18,4 +25,20 @@ public void setUniCashFilePath_nullPath_throwsNullPointerException() {
...
+        changedGuiSettings.setGuiSettings(new GuiSettings(0, 0, 0, 0));
+        assertNotEquals(changedGuiSettings, userPrefs);
+
+        var changedUniCashFilePath = new UserPrefs();
```
Comment (sp4ce-cowboy) [inline src/test/java/seedu/address/model/UserPrefsTest.java:39]:
```suggestion
        UserPrefs changedUniCashFilePath = new UserPrefs();
```

File: src/main/java/seedu/address/commons/util/ConfigUtil.java [hunk 20]
```diff
@@ -12,6 +12,10 @@
  */
 public class ConfigUtil {
 
+    private ConfigUtil() {
```
Comment (sp4ce-cowboy) [inline src/main/java/seedu/address/commons/util/ConfigUtil.java:15]: Great idea to add private default constructors to this and the surrounding classes

File: src/test/java/seedu/address/logic/commands/EditTransactionCommandTest.java [hunk 21]
```diff
@@ -154,24 +152,26 @@ public void equals() {
...
-        assertFalse(standardCommand.equals(null));
+        assertNotEquals(null, standardCommand);
 
         // different types -> returns false
-        assertFalse(copyDescriptor.equals(standardCommand));
-        assertFalse(standardCommand.equals(new ClearCommand()));
+        assertNotEquals(copyDescriptor, standardCommand);
+        assertNotEquals(standardCommand, new ClearTransactionsCommand());
+
+        assertFalse(standardCommand.equals(copyDescriptor));
```
[the remark this answers is not in the document]
Comment (woojiahao) [reply src/test/java/seedu/address/logic/commands/EditTransactionCommandTest.java:168]: `assertFalse` is used to explicitly test the

```java
if (!(other instanceof X)) {
  return false;
}
```

For some reason, CodeCov does not detect `assertNotEquals` to work. For the others, it works normally

**Review by sp4ce-cowboy — APPROVED** [review]
Just some minor syntax considerations and miscellaneous clarifications. Since the changes are quite broad, might wanna wait for someone else to cross-examine and approve also. 

Not relevant to the review but moving forth we might want to consider if the naming scheme i.e. appending or prepending Transaction or UniCash to the commands is something we want to maintain or start phasing out, since we're already removed the transaction sub-folder and having simpler names would make things neater, but it would also incur another rounding of mass changing everything, though I don't think that would be as expansive as this one. 

But other than that, this is great work, LGTM!

File: src/test/java/seedu/address/AppParametersTest.java (hunk 7 above)
Comment (sp4ce-cowboy) [reply src/test/java/seedu/address/AppParametersTest.java:5]: Scratch that, I just saw the messages on the group, you can ignore this comment

File: src/main/java/seedu/address/model/Model.java (hunk 3 above)
Comment (jamz903) [reply src/main/java/seedu/address/model/Model.java:42]: If i'm not wrong, we are allowing for duplicate transactions, meaning that they will be allowed to input the same item in the same shop immediately after even if it's the "same". Transaction identity should be by index for now, but we can consider implementing UID() later on for sure

**Review by jamz903 — APPROVED** [review]
good work jiahao, lgtm! 👍

**Pull request merged**